be/src/connector/hive_connector.cpp

```diff
@@ -495,6 +526,9 @@ Status HiveDataSource::_init_scanner(RuntimeState* state) {
         file_path /= scan_range.relative_path;
         native_file_path = file_path.native();
     }
+    if (native_file_path.empty()) {
+        native_file_path = _hive_table->get_base_path() + scan_range.relative_path;
+    }
 
     const auto& hdfs_scan_node = _provider->_hdfs_scan_node;
     auto fsOptions =
```

Comment:
Use of magic numbers: In is_scan_range_indicate_const_column function, you're returning -1 as a special value. It's generally recommended to define such values as named constants for readability and maintenance.

Code duplication: The conditional code inside _init_tuples_and_slots method shares many common expressions between both branches. This repeats assignments to _partition_slots, _partition_index_in_chunk etc. Consider refactoring to avoid duplicated statements.